src/thunderbots/software/geom/rect.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+#include "geom/point.h"
+
+/**
+ * A rectangle.
+ */
+class Rect final
+{
+   public:
+    /**
+     * Creates a new Rect from two corners.
```

Comment:
Should have javadoc style comments (for the whole file)

Reply:
DOne